spec/controllers/users/verify_profile_phone_controller_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+
+RSpec.describe Users::VerifyProfilePhoneController do
+  include Features::LocalizationHelper
+
+  let(:has_pending_profile) { true }
+  let(:user) { create(:user) }
+  let(:profile_phone) { user.phone }
+  let(:phone_confirmed) { false }
+  let(:pii_attributes) { Pii::Attributes.new_from_hash(phone: profile_phone) }
+  let(:pending_profile) { build(:profile, phone_confirmed: phone_confirmed) }
+
+  before do
+    stub_sign_in(user)
+    decorated_user = stub_decorated_user_with_pending_profile(user)
+    allow(decorated_user).to receive(:needs_profile_phone_verification?).
+      and_return(has_pending_profile)
+    allow(decorated_user).to receive(:needs_profile_usps_verification?).and_return(false)
+    allow(controller).to receive(:after_sign_in_path_for).with(user).and_return(profile_path)
+    allow(controller).to receive(:decrypted_pii).and_return(pii_attributes)
+  end
+
+  describe '#index' do
+    context 'user has pending profile' do
+      context 'phone is not confirmed' do
+        it 'redirects to profile page' do
+          get :index
+
+          expect(response).to redirect_to(profile_url)
+        end
+      end
+
+      context 'phone is same as 2FA' do
```

Comment:
I don't see any difference between this test and the one above? Did you mean to only set the profile_phone equal to the user phone in this test, and not in the test above as well? That way, we can test each scenario independently if it makes a difference.